src/middlewared/middlewared/plugins/account.py

```diff
@@ -381,7 +397,7 @@ async def do_create(self, data):
                 self.logger.warn('Failed to update authorized keys', exc_info=True)
                 raise CallError(f'Failed to update authorized keys: {e}')
 
-        return pk
+        return await self.get_instance(pk)
```

Comment:
As much as I wish I could do this too, this is a breaking of API 2.0

Now I think that with the appearance of documentation, we may introduce API 2.1 where we'll fix all piled up things like this. We'll only document return schemas for API 2.1, and for API 2.0 we'll continue returning integers.

I am not sure how much of a change that will be, I am writing this only having reviewed the code so far, so it's just a proposal, let's investigate further.